test/parallel/test-child-process-spawn-args.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
```

Comment:
There is already a test for this behaviour, could you expand it instead of adding another test file? See `test-child-process-spawn-typeerror.js` (and yes, it's name doesn't well reflect is purpose anymore). I assume this change effects, or SHOULD effect, every child_process API  that accepts an `argv` argument, showing that they have identical handling of the `argv` option, which is what the test I mentioned is set up for.

Reply:
@sam-github Nit or requirement to land?

Reply:
I'd rather it lands with these changes. Particularly with API consistency.

Reply:
That's a little bit controversial.

This file contains lots of unrelated checks, like https://github.com/nodejs/node/blob/master/test/parallel/test-child-process-spawn-typeerror.js#L115. Why do we test `execFile` function within `spawn` test file, shouldn't it placed in test-child-process-exec-file.js or something like that ?

And we have already separate test file test-child-process-spawn-argv0.js for testing argv0 param, so what the problem with `args` ?

In my opinion, *-spawn-typeerror.js should contain tests just about incorrect types, and just for spawn method.

Reply:
In addition, that fix effects only `spawn` and `spawnSync` methods.

Reply:
I also favor smaller, more-focused test files, especially for `child_process` tests in `parallel`.

Reply:
I consider that tests that are intended to ensure multiple APIs have behave identically, when they should, are best contained within a single file.

At this time, I can't tell from the tests whether all the child_process APIs do, in fact, have identical arg handling. Also, I can't tell if this PR makes them more (or less) consistent. Its possible that it fixes an inconsistency. Its possible it makes them inconsistent. I don't know. Ideally, the test structure would make that very clear. Every PR doesn't have to be ideal. I'm removing my review.